src/aspire/source/__init__.py

```diff
@@ -1,110 +1,163 @@
+import os.path
 import logging
 import numpy as np
+import pandas as pd
 
-from aspire.image import im_filter, im_translate
+from aspire.image import Image
 from aspire.volume import im_backproject, vol_project
-from aspire.utils.filters import IdentityFilter, ScalarFilter
+from aspire.utils.filters import ScalarFilter
 from aspire.estimation.noise import WhiteNoiseEstimator
 from aspire.utils import ensure
-from aspire.utils.coor_trans import grid_2d
-from aspire.utils.matlab_compat import m_reshape, randn, randi
+from aspire.utils.coor_trans import grid_2d, angles_to_rots
+from aspire.utils.matlab_compat import m_reshape, randn
+from aspire.io.starfile import Starfile
 
 logger = logging.getLogger(__name__)
 
 
-# TODO: The SourceFilter class shouldn't exist!
-# The assignment of filters to Images should happen inside the ImageSource object!
-class SourceFilter:
-    """
-    An object representing Filter objects and their assignments to a particular Source object
-    """
-    def __init__(self, filters, indices=None, n=None):
-        """
-        :param filters: An iterable of Filter objects.
-        :param indices: An iterable of indices representing the 0-indexed indices of an image stack
-            on which to apply the filters. If unspecified, `n` must be supplied, and individual filters are applied
-            randomly.
-        :param n: An integer representing the depth of the image stack on which this SourceFilter is applied.
-            Not needed if `indices` are supplied.
+class ImageSource:
+
+    # ----------------------------------------------------
+    # Class Attributes that can be overridden by subclasses
+    # ----------------------------------------------------
+
+    # Optional renaming of metadata fields, used
+    # These are used ONLY during serialization/deserialization (to/from Starfiles, for example).
+    _metadata_aliases = {
+        '_image_name':  '_image_name',
```

Comment:
I think it's more straightforward to have leading underscore in metadata fields, since that's what all fields in the STAR file need to have. Not having underscores and introducing/eliminating them magically while serializing/deserializing would be somewhat arbitrary.